consensus/src/terminator.rs

```diff
@@ -0,0 +1,141 @@
+use std::collections::HashMap;
+
+use futures::channel::oneshot::{channel, Receiver, Sender};
+use log::debug;
+
+type TerminatorConnection = (Sender<()>, Receiver<()>);
+type ShutdownConnection = (Receiver<()>, Option<TerminatorConnection>);
```

Comment:
Now you only use it once, as an argument to a method, making the calls less readable, so preferably just remove and split the arguments.